addons/website/static/src/js/editor/html_editor.js

```diff
@@ -0,0 +1,173 @@
+import { LinkPopover } from "@html_editor/main/link/link_popover";
+import { rpc } from "@web/core/network/rpc";
+import { _t } from "@web/core/l10n/translation";
+import { AutoComplete } from "@web/core/autocomplete/autocomplete";
+import { patch } from "@web/core/utils/patch";
+import { useAutofocus, useChildRef } from "@web/core/utils/hooks";
+import { loadAnchorsLinkPopover } from "../utils";
+
+/**
+ * The goal of this patch is to handle the URL autocomplete in the LinkPopover
+ * component. The URL autocomplete is used to suggest internal links, anchors.
+ * Before, the autocomplete was implemented as another OWL app. Now with this
+ * patch, URL autocomplete is implemented as a child component.
+ */
+
+/**
+ * this class is used to create a new autocomplete component that will be used
+ * in the LinkPopover component. Similar with AutoCompleteWithPages but it has
+ * two new props:
+ * - inputClass: to change the style of the input element in autocomplete
+ * - updateValue: to update the URL of the link element
+ */
+export class AutoCompleteInLinkpopover extends AutoComplete {
+    static props = {
+        ...AutoComplete.props,
+        inputClass: { type: String, optional: true },
+        updateValue: { type: Function, optional: true },
+    };
+    static template = "website.AutoCompleteInLinkpopover";
+
+    // overwrite the div class to avoid breaking the popover style
+    get autoCompleteRootClass() {
+        return `${super.autoCompleteRootClass} col`;
+    }
+    // apply classes on the input element in autocomplete
+    get inputClass() {
+        let classList = "o_input pe-3";
+        if (this.props.inputClass) {
+            classList = this.props.inputClass;
+        }
+        return classList;
+    }
+    /**
+     *
+     * @param indices
+     * @return {boolean}
+     * @private
+     */
+    _isCategory(option) {
+        return !!option?.separator;
+    }
+
+    getOption(indices) {
+        const [sourceIndex, optionIndex] = indices;
+        return this.sources[sourceIndex]?.options[optionIndex];
+    }
+
+    /**
+     * @override
+     */
+    onOptionMouseEnter(indices) {
+        if (!this._isCategory(this.getOption(indices))) {
+            return super.onOptionMouseEnter(...arguments);
+        }
+    }
+
+    /**
+     * @override
+     */
+    onOptionMouseLeave(indices) {
+        if (!this._isCategory(this.getOption(indices))) {
+            return super.onOptionMouseLeave(...arguments);
+        }
+    }
+    isActiveSourceOption(indices) {
+        if (!this._isCategory(this.getOption(indices))) {
+            return super.isActiveSourceOption(...arguments);
+        }
+    }
```

Comment:
The parent implementation always returns a boolean, you don't anymore when this if is falsy. It's a detail but it would be better to make sure the return type is kept consistent :pray: